File: React-Native/react-native-biometrics-backed.go
package main

import (
	"crypto"
	"crypto/rsa"
	"crypto/sha256"
	"crypto/x509"
	"encoding/base64"
	"log/slog"
)


func verifySignaturew(signature, publicKey, payload string) (error) {
	pubKeyBytes, err := base64.StdEncoding.DecodeString(publicKey)
	if err != nil {
		slog.Error("error decoding public key: " + err.Error())
		return err
	}

	pubKey, err := x509.ParsePKIXPublicKey(pubKeyBytes)
	if err != nil {
		slog.Error("error parsing public key: " + err.Error())
		return err
	}

	rsaPubKey, ok := pubKey.(*rsa.PublicKey)
	if !ok {
		slog.Error("error: not an RSA public key")
		return err
	}

	payloadBytes := []byte(payload)
	sigBytes, err := base64.StdEncoding.DecodeString(signature)
	if err != nil {
		slog.Error("error decoding signature: " + err.Error())
		return err
	}

	hashed := sha256.Sum256(payloadBytes)
	if err = rsa.VerifyPKCS1v15(rsaPubKey, crypto.SHA256, hashed[:], sigBytes);err != nil {
		slog.Error("signature verification failed: " + err.Error())
	}

	return nil
}

File: React-Native/react-native-biometrics-backed.js
// npm i node-rsa
const NodeRSA = require("node-rsa");


const verify = (payload, signature, publicKey) => {
    try {
        const publicKeyBuffer = Buffer.from(publicKey, "base64");
        const buffer = Buffer.from(payload);
        
        const key = new NodeRSA();
        const signer = key.importKey(publicKeyBuffer, "public-der");
        const verified = signer.verify(buffer, signature, "utf8", "base64");
        return verified;
    } catch (error) {
      // handle error
        console.log("error: ", error)
    }
}
